File: Manuel/2025-07-31/큐/queue.cpp
#include <iostream>
#include <string>
#include <queue>

#define endl "\n"

using namespace std;

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    int N; // 명령의 수
    cin >> N;

    queue<int> q; // int 타입을 저장하는 큐 선언

    for(int i = 0; i < N; i++)
    {
        string op; // 명령어를 저장할 문자열
        cin >> op;

        if(op == "push")
        {
            int X;
            cin >> X;
            q.push(X); // 정수 X를 입력받고 큐에 넣기
        }
        else if(op == "pop")
        {
            if(q.empty()) // 큐가 비어 있는지 확인
            {
                cout << -1 << endl;
            }
            else
            {
                cout << q.front() << endl; // 맨 앞 요소 확인
                q.pop(); // 맨 앞 요소 제거
            }
        }
        else if(op == "size")
        {
            cout << q.size() << endl; // 큐 안의 정수 개수 출력
        }
        else if(op == "empty")
        {
            if(q.empty())
            {
                cout << 1 << endl;
            }
            else
            {
                cout << 0 << endl;
            }
        }
        else if(op == "front")
        // 큐의 가장 앞에 있는 정수 출력
        {
            if(q.empty()) // 큐가 비어있는지 확인
            {
                cout << -1 << endl;
            }
            else
            {
                cout << q.front() << endl; // 맨 앞 요소 확인 (제거는 하지 않음!)
            }
        }
        else if(op == "back")
        // 큐에 가장 뒤 정수 출력
        {
            if(q.empty()) // 큐가 비어있는지 확인
            {
                cout << -1 << endl;
            }
            else
            {
                cout << q.back() << endl; // 맨 뒤 요소 확인 (제거하지 않음)
            }
        }
    }

    return 0;
}

File: Manuel/2025-07-31/큐/queue.py
import sys
from collections import deque

def solve():
    # 명령의 수
    N = int(sys.stdin.readline())
    # 큐로 사용할 deque 촉화
    queue = deque()

    # N개의 명령 처리
    for _ in range(N):
        op = sys.stdin.readline().split() # 명령을 공백으로 분리하여 읽기
        
        # 명령의 종류에 따라 다르게 처리
        if op[0] == 'push':
            X = int(op[1])
            queue.append(X)
        elif op[0] == 'pop':
            if queue: # 큐가 비어있지 않으면
                print(queue.popleft()) # 큐의 앞 요소를 제거하고 출력
            else: # 비어 있다면
                print(-1)
        elif op[0] == 'size':
            print(len(queue)) # 큐의 길이 (큐에 들어있는 정수의 개수) 출력
        elif op[0] == 'empty':
            if not queue: # 큐가 비어 있으면
                print(1)
            else:
                print(0)
        elif op[0] == 'front':
            if queue: # 큐가 비어 있지 않으면
                print(queue[0]) # 큐의 맨 앞 요소 확인
            else:
                print(-1)
        elif op[0] == 'back':
            if queue: # 큐가 비어 있지 않으면
                print(queue[-1]) # 큐의 맨 뒤 요소 확인
            else:
                print(-1)

solve()
